src/shaka_scheme/system/base/Data.hpp

```diff
@@ -36,7 +37,8 @@ class PrimitiveFormMarker;
 
 class Data;
 
-using NodePtr = std::shared_ptr<Data>;
+//using NodePtr = std::shared_ptr<Data>;
+using NodePtr = gc::GCNode;

```

Comment:
Comment: In the future, perhaps it may be helpful to have a dummy class that will automatically convert itself back out to the base type. C macro? Not a pretty solution but it may be one...